### PR TITLE
[#16] Create core data types

### LINKED ISSUE #16: Create core data types
`Inspection` and `Observation`.
See [Glossary](https://github.com/kowainik/stan/wiki/Glossary) for more info about the meaning.

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #16

### CODE REVIEW

File: src/Stan/Hie/Debug.hs [hunk 1]
```diff
@@ -1,6 +1,7 @@
-{-# LANGUAGE StandaloneDeriving #-}
```
Comment (chshersh) [inline src/Stan/Hie/Debug.hs:1]: Thanks for fixing the style :pray:

File: src/Stan/Inspection.hs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+-- | A type of the inspection.
+newtype Category = Category
+    { unCategory :: Text
+    } deriving newtype (Show)
+
+-- | Severity level of the inspection.
+data Severity
+    = Severe
+    | NotReallySevere
```
Comment (chshersh) [inline src/Stan/Inspection.hs:41]: :rofl:

File: src/Stan/Inspection.hs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+{- | Data type that represents a check/test, or how we call it
+__inspection__ that is provided by the Stan tool.
+-}
+data Inspection = Inspection
+    { inspectionId          :: !(Id Inspection)
+    , inspectionName        :: !Text
+    , inspectionDescription :: !Text
+    , inspectionSolution    :: !Text
+    , inspectionCategory    :: ![Category]
```
Comment (chshersh) [inline src/Stan/Inspection.hs:29]: Should it be `NonEmpty` list of categories? :thinking: Or is it possible not to have any categories?

File: src/Stan/Observation.hs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import Stan.Core.Id (Id)
+import Stan.Inspection (Inspection)
+
+
+{- | Data type to represent discovered by Stan vulnerabilities.
+-}
+data Observation = Observation
+    { observationId           :: !(Id Observation)
+    , observationInspectionId :: !(Id Inspection)
+    , observationLoc          :: !SrcSpan
```
Comment (chshersh) [inline src/Stan/Observation.hs:27]: Nice idea to store `SrcSpan` here! Do you think in the `Analysis` data type we will store something like `Map FilePath Observation` or `[(FilePath, Observation)]` for better grouping of all observations per file?

https://github.com/kowainik/stan/blob/e662cae85056ffcc0b7b353af87b0758911cccf9/src/Stan/Analysis.hs#L24

File: src/Stan/Inspection.hs (hunk 3 above)
Comment (vrom911) [reply src/Stan/Inspection.hs:29]: Oh, nice catch. I wouldn't like to have something uncategorised. This information can be quite valuable and important for filtering and searching. What do you think?

File: src/Stan/Observation.hs (hunk 4 above)
Comment (vrom911) [reply src/Stan/Observation.hs:27]: Good idea. Need to think more where the filepath would be more suitable to have. But I could imagine that the tuple (or separate data type) would be more convenient here.

**Pull request merged**